R/checkers.R

```diff
@@ -45,3 +57,268 @@ check_elements_non_neg <- function(x, arg = "x", call = rlang::caller_env()) {
   }
   invisible()
 }
+
+#' Check there is no missignness in a particular vector
+#'
+#' @param x the vector to check
+#' @param arg the name of the vector to check
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#'
+#' @return NULL, invisibly
+check_non_missingness <- function(x, arg = "x", call = rlang::caller_env()) {
+  any_missing <- any(is.na(x))
+  if (any_missing) {
+    cli::cli_abort(
+      c("{.arg {arg}} has missing values",
+        "!" = "All elements of{.arg {arg}} should be present",
+        "i" = "Element(s) {.val {which(is.na(x))}} are missing"
+      ),
+      class = "wwinference_input_data_error",
+      call = call
+    )
+  }
+  invisible()
+}
+
+#' Check that there are no repeated elements in the vector
+#'
+#' @description
+#' This function  checks that the elements of a vector are
+#' not repeated.
+#'
+#' @param x the vector to check
+#' @param arg the name of the vector to check
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#'
+#' @return NULL, invisibly
+check_for_repeat_elements <- function(x, arg = "x",
+                                      call = rlang::caller_env()) {
+  duplicates <- duplicated(x)
+  if (sum(duplicates) > 0) {
+    cli::cli_abort(
+      c("{.arg {arg}} has more than one element",
+        "i" = c(
+          "Multiple {.arg {arg}} are not currently supported.",
+          "Check that data is from a single location, and if so, provide",
+          "a single count data stream for the population in that location"
+        ),
+        "!" = "Duplicate element(s) index: {.val {which(duplicates)}}"
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+
+
+#' Check that the arguments are either integers or characters
+#'
+#' @description
+#' This is for unique identifiers of groupings, which we will allow to
+#' either be character or integers.
+#'
+#'
+#' @param x Object with type checking
+#' @param arg Name of the argument supplying the object
+#' @param call Calling environment to be passed to [cli::cli_abort()]
+#'
+#' @return NULL, invisibly
+check_int_or_char <- function(x, arg = "x", call = rlang::caller_env()) {
+  int_or_char_check_result <- checkmate::assert(
+    checkmate::check_integerish(x),
+    checkmate::check_character(x)
+  )
```

Comment:
This will not work as written. `checkmate::assert` combines checks into an assertion, whereas you want to combine them into another check. So this will need reworking.

Reply:
My bad, I kept forgetting this.

Reply:
This needs to be an OR not an AND so i removed the wrapper to `checkmate::assert()`